File: agent-cpp/src/agent/grenade_info.cc
#include "grenade_info.h"

#include <fmt/format.h>

#include <string>

namespace thuai7_agent {

auto format_as(GrenadeInfo const& object) -> std::string {
  return fmt::format("GrenadeInfo{{throwTick: {}, evaluatedPosition: {}}}",
                     object.throwTick, object.evaluatedPosition);
}

}  // namespace thuai7_agent

File: agent-python/agent/grenade_info.py
from typing import List, Literal

from .position import Position


class GrenadeInfo:
    def __init__(
        self,
        throwTick: int,
        evaluatedPosition: Position,
    ):
        self.throwTick = throwTick
        self.evaluatedPosition = evaluatedPosition

    def __str__(self) -> str:
        return f"GrenadeInfo{{throwTick: {self.throwTick}, evaluatedPosition: {self.evaluatedPosition}}}"
